### PR TITLE
Added support for missing architectures to BUILD.gn and fix incorrect repo URL

### PR DESCRIPTION
**Pull request by tmancey**
Fixes https://github.com/brave-intl/challenge-bypass-ristretto-ffi/issues/22

### CODE REVIEW

File: BUILD.gn [hunk 1]
```diff
@@ -8,16 +8,33 @@ challenge_bypass_build_path = "$root_gen_dir/challenge_bypass_ristretto/out"
...
+  if (target_cpu == "x86") {
+    challenge_bypass_lib_path += "/x86_64-linux-android"
+    challenge_bypass_platform = "Android x86_64"
+  } else {
+    challenge_bypass_lib_path += "/arm-linux-androideabi"
+    challenge_bypass_platform = "Android"
+  }
 } else if (is_ios) {
-  challenge_bypass_lib_path += "/x86_64-apple-ios"
+  if (target_cpu != "x64") {
```
Comment (bridiver) [inline BUILD.gn]: I think it's arm/arm64 for both ios and and android

Comment (tmancey) [reply BUILD.gn]: if x64 we build for simulator and if not x64 use the latest aarch64-apple-ios target. Brave Browser is iOS 12 and above so we can safely use this target

Comment (tmancey) [reply BUILD.gn]: Android we do use arm-linux-androideabi unless the target is x86

Comment (tmancey) [reply BUILD.gn]: @bridiver I have improved logic to use arm and arm64

**Review by bridiver — DISMISSED** [review]

File: BUILD.gn [hunk 2]
```diff
@@ -8,16 +8,33 @@ challenge_bypass_build_path = "$root_gen_dir/challenge_bypass_ristretto/out"
...
   challenge_bypass_lib_path += "/x86_64-apple-darwin"
+  challenge_bypass_platform = "macOS"
 } else if (is_linux) {
-  challenge_bypass_lib_path += "/x86_64-unknown-linux-musl"
+  challenge_bypass_lib_path += "/x86_64-unknown-linux-gnu"
+  challenge_bypass_platform = "Linux"
 } else if (is_android) {
-  challenge_bypass_lib_path += "/x86_64-linux-android"
+  if (target_cpu == "arm" || target_cpu == "arm64") {
+    challenge_bypass_lib_path += "/arm-linux-androideabi"
```
Comment (bridiver) [inline BUILD.gn]: is this right? Shouldn't one be arm and one be arm64? I guess @serg would be the best person to ask

Comment (tmancey) [reply BUILD.gn]: I did ask @serg and got him to confirm, but I have now added arm, arm64, x86 and x86_64 to future proof

**Review by bridiver — DISMISSED** [review]

**Review by bridiver — APPROVED** [review]

**Review by bbondy — APPROVED** [review]

**Pull request merged**